src/wallet/crypter.cpp

```diff
@@ -135,13 +136,31 @@ static bool DecryptKey(const CKeyingMaterial& vMasterKey, const std::vector<unsi
     return key.VerifyPubKey(vchPubKey);
 }
 
+static bool DecryptSpendingKey(const CKeyingMaterial& vMasterKey, const std::vector<unsigned char>& vchCryptedSecret, const libzcash::PaymentAddress& address, libzcash::SpendingKey& key)
+{
+    CKeyingMaterial vchSecret;
+    if(!DecryptSecret(vMasterKey, vchCryptedSecret, address.GetHash(), vchSecret))
+        return false;
+
+    if (vchSecret.size() != 32)
+        return false;
+
+    // TODO does this undo the benefits of using CKeyingMaterial?
+    std::vector<unsigned char> serialized(vchSecret.begin(), vchSecret.end());
+    CDataStream ss(serialized, SER_NETWORK, PROTOCOL_VERSION);
+    ss >> key;
+    return key.address() == address;
```

Comment:
``key`` will have been updated even if the address doesn't match; is this what we want?